### ISSUE TITLE
Show error if C++ function has default arguments.

### ISSUE THREAD

**Issue by alyst** [r-lib/cpp11#221]
I think it is fine that the default arguments are not supported (#4, #162).
But it is a bit confusing that there are no diagnostic messages about such functions, and the problem surfaces only at the linking time (also the linker errors about missing functions are cryptic).

Would be nice to stop with an error if default arguments are encountered.

**Comment by jimhester** [comment]
I agree, having a more informative error during registration would help a lot.
